ethclient/ethclient.go

```diff
@@ -133,7 +144,12 @@ func DialContext(ctx context.Context, rawurl string) (Client, error) {
 
 // NewClient creates a client that uses the given RPC client.
 func NewClient(c *rpc.Client) Client {
-	return &client{c}
+	return &client{c, nil}
```

Comment:
nit use named fields to reduce confusion and be able to move fields around later

```suggestion
	return &client{
		c: c,
	}
```